### DISCUSSION TITLE
import error autollm

### DISCUSSION THREAD

**Discussion by santhosh-sp** [safevideo/autollm#158] [Q&A] [answered]
Hi Team,

I tried to install autollm and imported it in venv (python3.8). I am getting below error message. Please check with this. 

**from autollm import AutoQueryEngine**

`    class AgentAction(Serializable):
  File "pydantic/main.py", line 198, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/fields.py", line 506, in pydantic.fields.ModelField.infer
  File "pydantic/fields.py", line 436, in pydantic.fields.ModelField.__init__
  File "pydantic/fields.py", line 552, in pydantic.fields.ModelField.prepare
  File "pydantic/fields.py", line 668, in pydantic.fields.ModelField._type_analysis
  File "/usr/lib/python3.8/typing.py", line 774, in __subclasscheck__
    return issubclass(cls, self.__origin__)
TypeError: issubclass() arg 1 must be a class`

Python: 3.8 
NVIDIA-SMI 520.61.05    Driver Version: 520.61.05    CUDA Version: 11.8

GPU 0: NVIDIA GeForce RTX 3090 (UUID: GPU-e2ce5f43-3889-5510-ec0c-831eab8c6ff0)

Please check this. 

Thanks in advance.
Santhosh

**Comment by kadirnar** [comment, marked as answer]
> Hi Team,
> 
> I tried to install autollm and imported it in venv (python3.8). I am getting below error message. Please check with this.
> 
> **from autollm import AutoQueryEngine**
> 
> ` class AgentAction(Serializable): File "pydantic/main.py", line 198, in pydantic.main.ModelMetaclass.__new__ File "pydantic/fields.py", line 506, in pydantic.fields.ModelField.infer File "pydantic/fields.py", line 436, in pydantic.fields.ModelField.__init__ File "pydantic/fields.py", line 552, in pydantic.fields.ModelField.prepare File "pydantic/fields.py", line 668, in pydantic.fields.ModelField._type_analysis File "/usr/lib/python3.8/typing.py", line 774, in __subclasscheck__ return issubclass(cls, self.__origin__) TypeError: issubclass() arg 1 must be a class`
> 
> Python: 3.8 NVIDIA-SMI 520.61.05 Driver Version: 520.61.05 CUDA Version: 11.8
> 
> GPU 0: NVIDIA GeForce RTX 3090 (UUID: GPU-e2ce5f43-3889-5510-ec0c-831eab8c6ff0)
> 
> Please check this.
> 
> Thanks in advance. Santhosh

I have tested. It works. Can you write library versions? Did you do this?

```pip install autollm autollm[readers]```
